refactor!: remove multi-world support

skaldarnar opened the pull request:
### Contains

Remove code and abstractions related to mutli-word support during the "Advanced" game creation flow.

### Context

The _multi-world feature_ was added as a GSOC project back in 2018 by @TheFlash98. For more information, checkout the original forum post [GSOC 2018 - Multiple Worlds][gsoc-2018-forum] or the [project page].

[gsoc-2018-forum]: https://forum.terasology.org/threads/gsoc-2018-multiple-worlds.2168/#post-15968
[project page]: https://summerofcode.withgoogle.com/archive/2018/projects/6321608952119296

While the world generation flow now supports to "create" multiple worlds and configure them separately, these multiple worlds were never experianceable in-game.

### How to test

Create a new game via the "Advanced" game setup flow.

### Outstanding before merging

- [X] disallow adding more than one world
- [X] replace world lists with single `WorldSetupWrapper`s in screen classes
- [x] remove dropdown related code in universe setup screen
- [x] remove remaining dropdown related code
- [x] remove selected world label, deduce selected world from world generator selection
- [x] don't include "Height Map" world generator as possible selection

### Follow-ups

- merge screens (except for world config screen)
- merge world config screen in

skaldarnar committed:
remove PreviewWorldScreen

jdrueckert committed:
refactor: remove WorldSetupWrapper list from StartPlayingScreen logic

jdrueckert committed:
refactor: selectedWorld as WorldSetupWrapper instead of String

jdrueckert committed:
refactor: remove world list from WorldPreGenerationScreen.java

jdrueckert committed:
refactor: disallow adding more than one world

jdrueckert committed:
refactor: replace world list with single WorldSetupWrapper

jdrueckert committed:
refactor: remove world counter

jdrueckert committed:
refactor: world into selectedWorld

jdrueckert committed:
refactor: remove copyOfSelectedWorld

jdrueckert commented:
@skaldarnar refactoring steps so far remove the option to add more than one world and removes respective code, in particular the former world lists are now only single WorldSetupWrappers...
I think a good next step would be to adjust the screens by replacing the world dropdowns by simple fields. That should allow us to get rid of a bunch of ui-related code.

skaldarnar committed:
refactor: remove 'add' and 'config' logic from Universe Setup Screen

skaldarnar committed:
refactor(ui): remove unused elements via NUI editor

This removes UI widgest referencing the following translation keys:

- `engine:menu#game-worlds`
- `engine:menu#add`
- `engine:menu#config`

jdrueckert committed:
ui: remove multi-world warning

jdrueckert committed:
ui: adjust universe setup description message

skaldarnar committed:
refactor: mark UniverseSetupScreen#worldNames as deprecated

skaldarnar committed:
refactor: remove worlds dropdowns from WorldPreGeneration and WorldSetup screens

jdrueckert committed:
ui: adjust world pregeneration screen description message

jdrueckert committed:
i18n: internationalize world name label in world setup screen

skaldarnar committed:
chore: remove unused method UniverseSetupScreen#refreshWorldDropdown

skaldarnar committed:
fix: prevent 'CoreWorlds:heightMap' from being added to the list of world generators

jdrueckert commented on engine/src/main/java/org/terasology/engine/rendering/nui/layers/mainMenu/UniverseSetupScreen.java:
```diff
@@ -275,8 +225,9 @@ private void addNewWorld(WorldGeneratorInfo worldGeneratorInfo) {
+    @Deprecated
     public void refreshWorldDropdown(UIDropdownScrollable worldsDropdown) {
-        worldsDropdown.setOptions(worldNames());
+        // no-op
```
why is this still needed?

jdrueckert replied:
nvm, got removed in commit 20782e3a9cf6308b3af39e868009a2ce2e3cad93

jdrueckert commented on engine/src/main/resources/org/terasology/engine/assets/ui/menu/universeSetupScreen.ui:
```diff
@@ -242,4 +180,4 @@
     ]
   }
-}
+}
```
newline

jdrueckert commented on engine/src/main/java/org/terasology/engine/rendering/nui/layers/mainMenu/UniverseSetupScreen.java:
```diff
@@ -291,6 +291,7 @@ private void initAssets() {
      * in the drop-down.
      * @return A list of world names encoded as a String
      */
+    @Deprecated
```
why do we still need this?

jdrueckert commented on engine/src/main/resources/org/terasology/engine/assets/ui/menu/worldPreGenerationScreen.ui:
```diff
@@ -265,4 +263,4 @@
     ]
   }
-}
+}
```
newline

jdrueckert committed:
chore: fix checkstyle findings

jdrueckert committed:
Merge branch 'develop' into refactor/remove-multi-world

jdrueckert committed:
chore: fix checkstyle warning

jdrueckert committed:
Merge branch 'develop' into refactor/remove-multi-world

jdrueckert committed:
chore: remove no-longer used deprecated method

jdrueckert reviewed (approved):
@skaldarnar your changes look fine to me.
can you also review the changes I did so we have a two-way approval?

jdrueckert committed:
Merge branch 'develop' into refactor/remove-multi-world

skaldarnar commented on .idea/kotlinc.xml:
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
where does this file come from? should it be committed?

skaldarnar replied:
according to https://intellij-support.jetbrains.com/hc/en-us/articles/206544839-How-to-manage-projects-under-Version-Control-Systems it should be checked in :shrug:

skaldarnar commented on engine/src/main/java/org/terasology/engine/rendering/nui/layers/mainMenu/StartPlayingScreen.java:
```diff
@@ -70,19 +65,8 @@ public void initialise() {
 
             SimpleUri uri;
             WorldInfo worldInfo;
+            //TODO: if we don't do that here, where do we do it? or does the world not show up in the game manifest?
```
This is done as part of `GameManifestProvider#createGameManifest` which is called on the other code paths, I think.

https://github.com/MovingBlocks/Terasology/blob/1d831ead54b3883a149f3671d07819a3d43c4637/engine/src/main/java/org/terasology/engine/rendering/nui/layers/mainMenu/GameManifestProvider.java#L96

We can remove this TODO comment and the line below.
